### ISSUE TITLE
[no-magic-numbers] detect error for enum values

### ISSUE THREAD

**Issue by karlismelderis** [eslint/eslint#13476]
**Tell us about your environment**

* Node version: v12.16.1
* npm version: v6.14.6
* Local ESLint version: v7.4.0 (Currently used)
* Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**

@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

I picked up only relevant part of config:
```js
  "overrides": [
    {
      "files": ["*.ts"],
      "parser": "@typescript-eslint/parser",
      "plugins": ["@typescript-eslint"],
      "parserOptions": {
        "project": "./tsconfig.json"
      },
      "rules": {
        "no-magic-numbers": ["warn"],
     }
  }
]
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export enum ExitCode {
  success = 0,
  failure = 1,
}
```

```bash
npx eslint .
```

**What did you expect to happen?**

I expect not to receive warnings/errors for enum
Maybe this PR is missing?
https://github.com/typescript-eslint/typescript-eslint/pull/543/files

**What actually happened? Please include the actual, raw output from ESLint.**

warnings are shown:
   18:13  warning  No magic number: 0     no-magic-numbers
   19:13  warning  No magic number: 1     no-magic-numbers

**Are you willing to submit a pull request to fix this bug?**
I guess it's already there - https://github.com/typescript-eslint/typescript-eslint/pull/543/files

**Comment by yeonjuan** [comment]
HI @karlismelderis 
If you want to use 'no-magic-numbers' with typescript code base, you should use `@typescript-eslint/no-magic-number`. see [@typescript-eslint/no-magic-numbers](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-magic-numbers.md)

**Comment by karlismelderis** [comment]
Worked like a charm

would it make sense to reference @typescript-eslint/no-magic-numbers in no-magic-numbers? :)

**Comment by mdjermanovic** [comment]
I'm closing this issue because the problem is solved.

> would it make sense to reference @typescript-eslint/no-magic-numbers in no-magic-numbers? :)

I don't think we would be able to maintain references to extension rules in third-party plugins, but I believe we could update our FAQ or Getting Started docs with some references to the `typescript-eslint/typescript-eslint` repository.
